src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1230,8 +1243,7 @@
                :where [:= :certname certname]
                :order-by [[:producer_timestamp :desc]]
                :limit 1}]
-    (sql/with-query-results result-set (hcore/format query)
-      (:producer_timestamp (first result-set)))))
+    (:producer_timestamp (first (jdbc/query (hcore/format query))))))
```

Comment:
I tried to make sure I only used query if I knew the query had a (small) result limit, i.e. "limit 1", since I believe query is not lazy; otherwise, I used db-query-with-resultset.